test/tests/functional/pbs_schedule_indirect_resources.py

```diff
@@ -0,0 +1,123 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2019 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestSchedulingIndirectResources(TestFunctional):
+    """
+    Test Scheduler resolve indirect resources correctly
+    """
+
+    def config_complex_for_grouping(self, res, res_type='string', flag='h'):
+        """
+        Configure the PBS complex for node grouping test
+        """
+        # Create a custom resource
+        attr = {"type": res_type, "flag": flag}
+        self.server.manager(MGR_CMD_CREATE, RSC, attr, id=res)
```

Comment:
there's also a server.add_resource() which basically does a qmgr, with some additional checking, i think. Just wondering if we should be using that interface to create the resource. What do you think?

Reply:
It was server.add_resource() earlier but Bhroam said that that's an old interface and suggested to change it to this.

Reply:
Oh, interesting, @agrawalravi90 is right.  server.add_resource() does use manager().  It existed before qmgr modified resources and modified the resourcedef file itself.  I thought it still did.  I guess it got rewritten at some point.  The only difference is that it keeps the PTL cache up to date.